### DISCUSSION TITLE
There should be a progress indicator

### DISCUSSION THREAD

**Discussion by aljawaid** [creecros/Mailmagik#11] [Ideas]
mailboxes can be quite big... if the plugin is scanning the inbox to look for certain emails, it can take time and it does.... there should be a user progress notification somehow.

**Comment by creecros** [comment]
for the most part yes, one of the reasons the task email feature kind of sucks. my only thought that could eliminate that time, kinda sucks, because I would need to add a table, and really don't think it's worth the trouble. I also have no idea how to do a progress bar, like clueless.

**Comment by alfredbuehler** [comment]
I think, it would be sufficient if the screen changed **immediately** to the mail list, issue a "scanning your mailbox..." message and finally fills up the list. Actually, after clicking on task email link, there is no visual action.

**Reply by aljawaid** [reply]
Agreed, that's what I was also trying to explain

**Reply by creecros** [reply]
I definitely understand. how to do that, im not yet sure.

**Reply by creecros** [reply]
i spent all morning trying to figure that out. i failed.

**Reply by aljawaid** [reply]
What if you make a new page with a hook. Then update hook to show pending progress, then update the same hook when emails are found.

**Reply by creecros** [reply]
ya, i tried something similiar, ill look again with fresh eyes later. sometimes its hard to find a solution when you get stuck on something.

**Reply by creecros** [reply]
i think my issue was, i couldn't figure out how to go from template back to controller, without a link. no redirect, and apparently i havent ever done anything like that.

**Reply by aljawaid** [reply]
> i think my issue was, i couldn't figure out how to go from template back to controller, without a link. no redirect, and apparently i havent ever done anything like that.

What about that meta refresh tag which autoloads the page. But then I guess you don't know how long it would take to scan the mails. I used a busy inbox and it took ages, like around 60secs

**Comment by alfredbuehler** [comment]
https://github.com/creecros/kbphpimap/commit/ba62f1d08fe24b8d41e07d14ae6b532418f01507

I've applied some changes to the mail-scanning, for Task-Emails. Instead of getting all mails and then dropping the non-task mails, the Task#-prefix is now checked on the server, and only matching mails will be loaded.  This should bring some more speed on processing, especially when there are plenty of other mails around in the inbox.

**Reply by creecros** [reply]
Good stuff!

**Comment by creecros** [comment]
https://github.com/creecros/kbphpimap/commit/837cf79cdd23bb9c0380c227f7b919925ff612c3

somehow this worked. it aint pretty.

**Reply by alfredbuehler** [reply]
What exactly?
Ah, I see, you're trying a progress indicator.

**Reply by creecros** [reply]
i would call it more of a something is happening indicator, its definately not going to track progress, but it will at least let people know stuff is going down

**Reply by alfredbuehler** [reply]
Good luck! For myself, I gave up…

**Reply by creecros** [reply]
As far as i can tell it works:
![Peek 2023-02-07 16-01](https://user-images.githubusercontent.com/26339368/217364815-1f2e2162-955a-433d-bb39-b4904331c617.gif)

**Comment by alfredbuehler** [comment]
Thank you, looks great! Maybe you could put the phrase “Scanning mailbox” somewhere?
